src_c/geometry.c

```diff
@@ -244,10 +244,51 @@ geometry_regular_polygon(PyObject *_null, PyObject *const *args,
     return (PyObject *)ret;
 }
 
+static PG_FORCE_INLINE void
+_normalize_rect(SDL_Rect *rect)
+{
+    if (rect->w < 0) {
+        rect->x += rect->w;
+        rect->w = -rect->w;
+    }
+    if (rect->h < 0) {
+        rect->y += rect->h;
+        rect->h = -rect->h;
+    }
+}
+
+static PyObject *
+geometry_rect_to_polygon(PyObject *_null, PyObject *arg)
+{
+    SDL_Rect rect, *tmp;
+    PyObject *ret;
+
+    if (!(tmp = pgRect_FromObject(arg, &rect))) {
+        if (PyErr_Occurred()) {
+            return NULL;
+        }
+        return RAISE(PyExc_TypeError, "rect must be a valid Rect object");
+    }
+
+    _normalize_rect(tmp);
+
+    double vertices[] = {tmp->x, tmp->y,          tmp->x + tmp->w,
+                         tmp->y, tmp->x + tmp->w, tmp->y + tmp->h,
+                         tmp->x, tmp->y + tmp->h};
```

Comment:
I don't know why but clang format made this strange arrangement and considers it valid.